src/main/java/baritone/pathing/movement/movements/MovementDiagonal.java

```diff
@@ -132,27 +132,33 @@ public static void cost(CalculationContext context, int x, int y, int z, int des
                 }
             }
         }
-        double multiplier = WALK_ONE_BLOCK_COST;
+        Block cuttingOver1 = context.get(x, y - 1, destZ).getBlock(); //cutting over adjacent x-axis block
+        Block cuttingOver2 = context.get(destX, y - 1, z).getBlock(); //cutting over adjacent z-axis block
+        if (MovementHelper.isLava(cuttingOver1) || MovementHelper.isLava(cuttingOver2)) {
+            return;
+        }
+        double multiplier;
+        if ((cuttingOver1 == Blocks.MAGMA || cuttingOver2 == Blocks.MAGMA) && !MovementHelper.hasFrostWalker(context)) {
+            multiplier = SNEAK_ONE_BLOCK_COST;
+        } else {
+            multiplier = WALK_ONE_BLOCK_COST;
+        }
         // For either possible soul sand, that affects half of our walking
         if (destWalkOn.getBlock() == Blocks.SOUL_SAND) {
             multiplier += (WALK_ONE_OVER_SOUL_SAND_COST - WALK_ONE_BLOCK_COST) / 2;
         } else if (destWalkOn.getBlock() == Blocks.WATER) {
             multiplier += context.walkOnWaterOnePenalty * SQRT_2;
+        } else if (destWalkOn.getBlock() == Blocks.MAGMA && (cuttingOver1 != Blocks.MAGMA || cuttingOver2 != Blocks.MAGMA) && !MovementHelper.hasFrostWalker(context)) {
```

Comment:
I think it should be `cuttingOver1 != magma && cuttingOver2 != magma`.
As soon as one of the two is magma (and there's no frostwalker boots) the multiplier is already `SEAK_ONE_BLOCK_COST`. So only if both are not magma it may be increased.